rfd/0236-access-list-preset.md

```diff
@@ -0,0 +1,263 @@
+- `reviewer`: A role that allows reviewing access requests to resources defined in the "access" roles. This role is directly assigned to owners of this access list (as owner grants).
+
+### User story: As an admin, I want to require a select group of users (local or Okta imported) to request for short-term access to resources
+
+The admin can create an access list using the preset `short-term`.
+
+Preset `short-term` represents an access list that utilizes JIT. Owners of the access list are reviewers. Members of the access list are requesters that are required to request access to resources and then upon approval are granted short-term access to requested Teleport resources.
+
+All admin needs to do for this preset is to define the metadata, owners, and members of the access list and define the access to resources (e.g. labels) and then Teleport will create the following type of roles and create/modify the access list accordingly:
+
+- `access`: Role(s) related to allowing access to resources. The access specs are determined by admin input. These roles are NOT directly assigned to anyone but instead are indirectly assigned to other roles created below.
+- `requester`: A role that allows requesting for the resources defined in the `access` roles. This role is directly assigned to members of this access list (as member grants).
+- `reviewer`: A role that allows reviewing access requests to resources defined in the `access` roles. This role is directly assigned to owners of this access list (as owner grants).
+
+### Web UI/UX
+
+The Web UI guide has two parts:
+
+#### Defining the `access_list` resource
+
+The guide will ask the admin for input that defines the owners, members, and metadata (e.g. title, description, audit frequency) of an access list. It is the same type of inputs asked of an admin currently in the web UI, but now these guides will break these inputs into smaller multi steps allowing to add clarifications for each step.
+
+#### Defining the allow specs of `role` resources
+
+Only a part of the `role` resource requires admin input and that is the [specs of a role](https://github.com/gravitational/teleport/blob/85829ad33c3af785a220efc0e8a8b46af102f039/api/proto/teleport/legacy/types/types.proto#L3537) specifically the `allow` field which defines giving access to resources.
+
+Instead of expecting the admin to define the `role` spec using traditional YAML or through a long form of input fields, the guide will provide a more [interactive UI/UX](https://www.figma.com/design/CzxaBHF8hirrYv2bTVa4Rw/Identity-Governance?node-id=7973-166743&t=A3qT6ANUJF67SSsr-0) way of defining the role specs. To the admin, the guide will just look like they are "defining access to resources", not defining a role. The fact that roles are being created/edited under the hood is hidden from the admin.
```

Comment:
I was just going to ask if the UI could have a preview of all resources that would be granted, then I opened that figma link, it looks even better than I was imagining